server/raft_transport.go

```diff
@@ -116,28 +117,42 @@ func (t *rpcTransport) Send(req *multiraft.RaftMessageRequest) error {
 		return err
 	}
 
-	client := rpc.NewClient(addr, nil, t.rpcContext)
-	select {
-	case <-client.Ready:
-	case <-client.Closed:
-		return util.Errorf("raft client failed to connect")
-	}
-
-	call := client.Go(raftMessageName, protoReq, &proto.RaftMessageResponse{}, nil)
-	select {
-	case <-call.Done:
-		// If the call failed synchronously, report an error.
-		return call.Error
-	default:
-		// Otherwise, fire-and-forget.
-		go func() {
-			<-call.Done
-			if call.Error != nil {
-				log.Errorf("raft message failed: %s", call.Error)
-			}
-		}()
-		return nil
-	}
+	// NewClient can block, so fire-and-forget from here on.
```

Comment:
Sorry, it can't block. It was the client never getting ready (or returning from a client.Go call, depending on when the remote died).